internal/controllers/extension_controller.go

```diff
@@ -18,30 +18,54 @@ package controllers
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
 
+	mmsemver "github.com/Masterminds/semver/v3"
 	"github.com/go-logr/logr"
 	catalogd "github.com/operator-framework/catalogd/api/core/v1alpha1"
+	"github.com/operator-framework/deppy/pkg/deppy/solver"
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+	kappctrlv1alpha1 "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/kappctrl/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/equality"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/utils/ptr"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
+	catalogsort "github.com/operator-framework/operator-controller/internal/catalogmetadata/sort"
+
 	ocv1alpha1 "github.com/operator-framework/operator-controller/api/v1alpha1"
+	"github.com/operator-framework/operator-controller/internal/catalogmetadata"
+	catalogfilter "github.com/operator-framework/operator-controller/internal/catalogmetadata/filter"
 	"github.com/operator-framework/operator-controller/pkg/features"
 )
 
 // ExtensionReconciler reconciles a Extension object
 type ExtensionReconciler struct {
 	client.Client
+	BundleProvider BundleProvider
+	Resolver       *solver.Solver
+	HasKappApis    bool
 }
 
+var errkappAPIUnavailable = errors.New("kapp-controller apis unavailable on cluster")
+
 //+kubebuilder:rbac:groups=olm.operatorframework.io,resources=extensions,verbs=get;list;watch;create;update;patch;delete
-//+kubebuilder:rbac:groups=olm.operatorframework.io,resources=extensions/status,verbs=update;patch
+//+kubebuilder:rbac:groups=olm.operatorframework.io,resources=extensions/status,verbs=get;update;patch
```

Comment:
There's no such thing as the `get` verb on the `/status` subresource. I removed this in one of the Extension API introduction followups. I'm guessing this was an easy thing to miss in the rebase.

Reply:
My bad! Yeah, missed it.